toolkit/tools/pkg/imagecustomizerlib/customizeoverlay.go

```diff
@@ -46,6 +46,9 @@ func updateGrubConfigForOverlay(imageChroot *safechroot.Chroot, overlays []image
 
 	// Iterate over each Overlay configuration
 	for _, overlay := range overlays {
+		if overlay.Partition == nil {
+			formattedPartition = ""
+		}
 		if overlay.Partition != nil {
 			formattedPartition, err = systemdFormatPartitionId(overlay.Partition.IdType, overlay.Partition.Id)
 			if err != nil {
```

Comment:
This can be simplified by initializing `formattedPartition = ""` as the default value outside the if statement and only changing the value if `overlay.Partition != nil`. So we only need the second if statement here.

Reply:
Fixed.